### ISSUE TITLE
inappropriate use of `.unshift()` for `this[train_symbol].unshift()`

### ISSUE THREAD

**Issue by chazkiker2** [Light-Labs/ryderserial-proto#3]
https://github.com/Light-Labs/ryderserial-proto/blob/1d478552f89d07a8ca4937cb7a0f7f5f36edaf35/src/index.js#L117

`unshift(...items: any[]): number` takes in arbitrarily many items, adds them to the beginning of an array, and returns the new LENGTH of the array. 

1. cannot de-structure `var [, resolve, reject]` from a number
2. calling `this[train_symbol].unshift()` without any parameters will leave `this[train_symbol]` unchanged and continue returning the same, unchanging length of that array.

@MarvinJanssen I found this while converting this file to TypeScript and am thinking about encapsulating some of `this[train_symbol]` into its own bundled class... thoughts?

Can you confirm that `this[train_symbol]` will be an array of entries like so: 

```typescript

type TrainEntry = [
  // data
  string,
  // resolve
  (value: any) => void,
  // reject
  (error: any) => void,
  // isEscapedByte
  boolean,
  // message ?
  string
]

this[train_symbol]: Array<TrainEntry>
```

**Comment by MarvinJanssen** [comment]
`RyderSerial.RESPONSE_LOCKED` is not currently emitted by the Ryder so that flow will (luckily) never be entered into. Still, not great to have faulty unused code in there... 

TypeScript conversion is a great idea. The last field of the train is an output buffer. 

This package will likely also benefit from a bunch of refactoring. It is in the pipeline.

**Comment by chazkiker2** [comment]
This was resolved in #5
